src/main/java/eu/stamp_project/mutationtest/descartes/operators/ThisMutationOperator.java

```diff
@@ -0,0 +1,48 @@
+package eu.stamp_project.mutationtest.descartes.operators;
+
+import org.pitest.classinfo.ClassName;
+import org.pitest.reloc.asm.MethodVisitor;
+import org.pitest.reloc.asm.Opcodes;
+import org.pitest.reloc.asm.Type;
+import org.pitest.reloc.asm.commons.Method;
+
+/**
+ * Replaces the method body with a <code>return null</code> statement
+ */
+public class ThisMutationOperator extends MutationOperator {
+
+    /**
+     * Returns a value indicating whether the operator can transform the given
+     * method. In this case whether <code>null</code> could be assigned to the
+     * return type.
+     *
+     * @param method Method to be tested by the operator
+     * @return A boolean value indicating if null can be assigned to the return type
+     */
+    @Override
+    public boolean canMutate(ClassName className, Method method) {
+        String classNameAsInternal = className.asInternalName();
+        String methodReturn = method.getReturnType().toString();
+        if (methodReturn.contains(classNameAsInternal)) {
```

Comment:
Why `methodReturn.contains(classNameAsInternal)`?

Reply:
This is because, `method.getReturnType()` contains `className.asJavaName`.

Reply:
Yes, sure, I get that. But, as we discussed before, two different classes that, somehow, one's name is the prefix of the other's name  will be considered as equal by your implementation. For `a.b.CD` and `a.b.C` the condition will be `true` but they are not the same class. So the question  really is Why aren't you comparing them for equality? The reason actual reason you are doing that, I suppose, is because `Method` will give you the descriptor of the type, something like `La/b/C;` while `ClassName` will return the internal name which will be `a/b/C`.  But in that case, either you convert the descriptor to the internal name or you transform the internal name into a descriptor and then you check that both are **equal**. Otherwise the comparison is wrong.

Reply:
I changed the code such that it compares both of them to be equal. Please look into the same!